src/main/java/mate/jdbc/service/DriverServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.jdbc.service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.jdbc.dao.DriverDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.model.Driver;
+
+public class DriverServiceImpl implements DriverService {
+
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver create(Driver driver) {
+        return driverDao.create(driver);
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        return Optional.of(driverDao.get(id)).orElseThrow(
```

Comment:
is here `Optional.of(Optional<Driver>)`?
```suggestion
        return driverDao.get(id).orElseThrow(
```

Reply:
No, here is Optional<Driver>. I have tested your requirment, but it doesn't work correct(

Reply:
why, if driver.get() returns Optional of Driver? and here it is wrapping in Optional once again. look once again in the task description, what `service.get()` should return?